gems/pending/VMwareWebService/MiqVimInventory.rb

```diff
@@ -1891,6 +1905,55 @@ def addStoragePodObj(spObj)
     addObjHash(:StoragePod, spObj)
   end
 
+  def pbmProfiles_locked
+    log_header = "MiqVimInventory.pbmProfiles_locked:"
+    raise "pbmProfiles_locked: cache lock not held" unless @cacheLock.sync_locked?
+    return @pbmProfiles if @pbmProfiles
+
+    $vim_log.info "#{log_header} loading pbmProfile cache for #{@connId}"
+    begin
+      @pbmProfiles = {}
+
+      @cacheLock.sync_lock(:EX) if (unlock = @cacheLock.sync_shared?)
```

Comment:
I always assumed that it doubled to protect the cache as well as prevent concurrent access to the session object (so you don't have two threads making API calls at the same time) but I could be wrong about that last bit, @roliveri?